Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/BlendGraphViewWidget.cpp

```diff
@@ -565,6 +594,55 @@ namespace EMStudio
         }
     }
 
+#if defined(EMFX_ANIMGRAPH_PROFILER_ENABLED)
+    void BlendGraphViewWidget::AddProfilingAction(const char* actionName, EOptionFlag optionFlag)
+    {
+        m_actions[optionFlag] = new QAction(
+            tr(actionName),
+            this);
+        m_actions[optionFlag]->setCheckable(true);
+        connect(m_actions[optionFlag], &QAction::triggered, this, [=]([[maybe_unused]] bool checked) { OnDisplayProfiling(optionFlag); });
+    }
+
+    void BlendGraphViewWidget::OnDisplayProfiling(EOptionFlag profileOption)
+    {
+        bool show = GetOptionFlag(profileOption);
+
+        uint32 displayFlags = m_parentPlugin->GetDisplayFlags();
+        switch (profileOption)
+        {
+        case VISUALIZATION_PROFILING_UPDATE:
+            displayFlags = AnimGraphPlugin::DISPLAYFLAG_PROFILING_UPDATE;
+            break;
+        case VISUALIZATION_PROFILING_TOPDOWN:
+            displayFlags = AnimGraphPlugin::DISPLAYFLAG_PROFILING_TOPDOWN;
+            break;
+        case VISUALIZATION_PROFILING_POSTUPDATE:
+            displayFlags = AnimGraphPlugin::DISPLAYFLAG_PROFILING_POSTUPDATE;
+            break;
+        case VISUALIZATION_PROFILING_OUTPUT:
+            displayFlags = AnimGraphPlugin::DISPLAYFLAG_PROFILING_OUTPUT;
+            break;
+        default:
```

Comment:
suggestion: minor defensive programming but it's good form to add a `break` after the `default` case just in case someone adds something after this in future - not 100% necessary but it's good hygiene

Reply:
who would be that insane to add more condition code after default?

Reply:
You never know 😉